test/Microsoft.ML.Tests/TextLoaderTests.cs

```diff
@@ -1040,87 +1040,68 @@ public void TestDifferentDecimalMarkersAtTheSameTime(bool useCorrectPeriod, bool
-                                Assert.Equal(Single.NaN, featuresCommaArray[i]);
-                            }
+                            // Check that none of the two files loadad NaNs
+                            // As both of them should have been loaded correctly
+                            Assert.Equal(featuresPeriodArray[i], featuresCommaArray[i]);
+                            Assert.NotEqual(Single.NaN, featuresPeriodArray[i]);
+                        }
+                        else if (!useCorrectPeriod && !useCorrectComma)
+                        {
+                            // Check that everything was loaded as NaN
+                            // Because the wrong decimal marker was used for both loaders
+                            Assert.Equal(featuresPeriodArray[i], featuresCommaArray[i]);
+                            Assert.Equal(Single.NaN, featuresPeriodArray[i]);
+                        }
+                        else if (!useCorrectPeriod && useCorrectComma)
+                        {
+                            // Check that only the file with commas was loaded correctly
+                            Assert.Equal(Single.NaN, featuresPeriodArray[i]);
+                            Assert.NotEqual(Single.NaN, featuresCommaArray[i]);
+                        }
+                        else
+                        {
+                            // Check that only the file with periods was loaded correctly
+                            Assert.NotEqual(Single.NaN, featuresPeriodArray[i]);
+                            Assert.Equal(Single.NaN, featuresCommaArray[i]);
                         }
                     }
                 }
-
-                // Check how many custom instances there are of TextLoader.ValueCreatorCache
-                var vccType = typeof(TextLoader).GetNestedType("ValueCreatorCache", BindingFlags.NonPublic | BindingFlags.Static);
-                var customInstancesInfo = vccType.GetField("_customInstances", BindingFlags.NonPublic | BindingFlags.Static);
-                var customInstancesObject = customInstancesInfo.GetValue(null);
-                var customInstancesCount = (int)customInstancesObject.GetType().GetProperty("Count").GetValue(customInstancesObject, null);
-                var customInstancesContainsMethod = customInstancesObject.GetType().GetMethod("ContainsKey");
-
-                // Regardless of useCorrectPeriod and useCorrectComma
-                // Since we always created a TextLoader with Comma as DecimalMarker
-                // There should always be 1, and only 1, custom instance of ValueCreatorCache, corresponding to the comma option
-                // Even after running multiple times the loop above.
-                Assert.Equal(1, customInstancesCount);
-                Assert.True((bool)customInstancesContainsMethod.Invoke(customInstancesObject, new[] { (object) DoubleParser.OptionFlags.UseCommaAsDecimalMarker }));
```

Comment:
I had to remove this part of this test I wrote on #5154, because in here I checked for an static member which is also modified by the new test I'm introducing on this PR. If that other new tests execute before this test I've modified in here, then the conditions these lines checked won't hold. So that's why I removed this and the for-loop that was above it.